nyaa/views/torrents.py

```diff
@@ -487,3 +487,21 @@ def _get_cached_torrent_file(torrent):
             out_file.write(torrents.create_bencoded_torrent(torrent, metadata_base))
 
     return open(cached_torrent, 'rb'), os.path.getsize(cached_torrent)
+
+
+@bp.route('/comments-preview/<int:torrent_id>/', methods=['GET'])
+def comments_preview(torrent_id):
+    torrent = (models.Torrent.query
+               .options(joinedload('comments'))
```

Comment:
While this won't kill performance, I'd much rather see it optimized to only fetch the N latest comments for a given torrent, using the query `.limit()` (instead of fetching them all and grabbing the last five).